iceoryx_hoofs/include/iceoryx_hoofs/error_reporting/error_reporting_macros.hpp

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2023 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_ERROR_REPORTING_API_HPP
+#define IOX_HOOFS_ERROR_REPORTING_API_HPP
+
+#include "iceoryx_hoofs/error_reporting/configuration.hpp"
+#include "iceoryx_hoofs/error_reporting/custom/error_kind.hpp"
+#include "iceoryx_hoofs/error_reporting/error_forwarding.hpp"
+
+#include "iceoryx_hoofs/error_reporting/source_location.hpp"
+
+// ***
+// * Define public error reporting API
+// ***
+
+// NOLINTBEGIN(cppcoreguidelines-macro-usage) source location requires macros
+
+// The following macros are statements (not expressions).
+// This is important, as it enforces correct use to some degree.
+// For example they cannot be used as function arguments and must be terminated with a ';'.
+//
+// Note: once source location becomes available without macro usage this could (and arguably should)
+// be transformed into a function API.
+
+/// @brief calls panic handler and does not return
+/// @param message message to be forwarded
+/// @note could actually throw if desired without breaking control flow asssumptions
+#define IOX_PANIC(message)                                                                                             \
+    do                                                                                                                 \
+    {                                                                                                                  \
+        iox::err::forwardPanic(CURRENT_SOURCE_LOCATION, message);                                                      \
+    } while (false)
```

Comment:
No. We will tackle this once it becomes an issue and use idiomatic macros for now to prevent using them as expressions.

Using cyclomatic complexity with hard thresholds is foolish anyway (and we do not).
Usually it is easy to spot bad functions with high complexity in reviews anyway.